rust-bins/src/bin/keygen-genesis.rs

```diff
@@ -0,0 +1,241 @@
+        help = "The integer identifying the identity provider"
+    )]
+    ip_identity: u32,
+    #[structopt(long = "name", help = "Name of the identity provider")]
+    name: String,
+    #[structopt(long = "url", help = "url to identity provider")]
+    url: String,
+    #[structopt(long = "description", help = "Description of identity provider")]
+    description: String,
+    #[structopt(
+        long = "bound",
+        help = "Upper bound on messages signed by the IP",
+        default_value = "30"
+    )]
+    bound: u32,
+    #[structopt(long = "out-pub", help = "File to output the public keys to.")]
+    out_pub: PathBuf,
+}
+
+#[derive(StructOpt)]
+struct KeygenAr {
+    #[structopt(long = "rand-input", help = "File with randomness.")]
+    rand_input: PathBuf,
+    #[structopt(
+        long = "ar-identity",
+        help = "The integer identifying the anonymity revoker"
+    )]
+    ar_identity: u32,
+    #[structopt(long = "name", help = "Name of the anonymity revoker")]
+    name: String,
+    #[structopt(long = "url", help = "url to anonymity revoker")]
+    url: String,
+    #[structopt(long = "description", help = "Description of anonymity revoker")]
+    description: String,
+    #[structopt(long = "out-pub", help = "File to output the public keys to.")]
+    out_pub: PathBuf,
+}
+
+#[derive(StructOpt)]
+#[structopt(
+    about = "Tool for generating keys for genesis",
+    author = "Concordium",
+    version = "0.2"
+)]
+enum KeygenTool {
+    #[structopt(name = "keygen-ip", about = "Generate identity provider keys")]
+    KeygenIp(KeygenIp),
+    #[structopt(name = "keygen-ar", about = "Generate anonymity revoker keys")]
+    KeygenAr(KeygenAr),
+}
+
+fn main() {
+    let app = KeygenTool::clap()
+        .setting(AppSettings::ArgRequiredElseHelp)
+        .global_setting(AppSettings::ColoredHelp);
+    let matches = app.get_matches();
+    let kg = KeygenTool::from_clap(&matches);
+    use KeygenTool::*;
+    match kg {
+        KeygenIp(kgip) => {
+            if let Err(e) = handle_generate_ip_keys(kgip) {
+                eprintln!("{}", e)
+            }
+        }
+        KeygenAr(kgar) => {
+            if let Err(e) = handle_generate_ar_keys(kgar) {
+                eprintln!("{}", e)
+            }
+        }
+    }
+}
+
+macro_rules! succeed_or_die {
+    ($e:expr, $match:ident => $s:expr) => {
+        match $e {
+            Ok(v) => v,
+            Err($match) => return Err(format!($s, $match)),
+        }
+    };
+    ($e:expr, $s:expr) => {
+        match $e {
+            Some(x) => x,
+            None => return Err($s.to_owned()),
+        }
+    };
+}
+
+fn handle_generate_ar_keys(kgar: KeygenAr) -> Result<(), String> {
+    let bytes_from_file = succeed_or_die!(fs::read(kgar.rand_input), e => "Could not read random input from provided file because {}");
+    if bytes_from_file.len() < 64 {
+        return Err("Provided randomness should be of size at least 64 bytes".to_string());
```

Comment:
I don't think we should have any such restrictions in this. In principle empty string is fine as a seed.